### ISSUE TITLE
Question about integrating extra information for disease prediction

### ISSUE THREAD

**Issue by HelloWorldLTY** [gerstung-lab/Delphi#31]
Hi, I want to explore more about how to improve disease risk prediction by combining delphi outputs and other baseline information.

My understanding is, the Extended Data Figure 5 utilizes PRS/other records + the prediction logits of targeted disease in each patient as training dataset, is this correct? Both PRS and records are personal level? Thanks a lot.

**Comment by HelloWorldLTY** [comment]
I also find an inconsistent part for the generate function of Delphi, in model.py.

```
    def generate(self, idx, age, max_new_tokens=100, max_age=85*365.25, no_repeat=True, termination_tokens=None, top_k=None):
```
The max_age has already been pre-defined in the function head, but in line 392,  the max age seems different:

```
 t_next = torch.clamp(-torch.exp(-logits) * torch.rand(logits.shape, device=idx.device).log(), min=0, max=365*80).min(1)
```

I think this max value should be equal to max_age, and thus we can modify it based on our own data's distrbution. 

And I also believe the prediction should be using gender+age+logit to predict the next event.

Please correct me if I am wrong, thank you.

**Comment by yozhikoff** [comment]
Hi @HelloWorldLTY !

> My understanding is, the Extended Data Figure 5 utilizes PRS/other records + the prediction logits of targeted disease in each patient as training dataset, is this correct? Both PRS and records are personal level? Thanks a lot.

Yes, exactly. Both are personal-level.

> I also find an inconsistent part for the generate function of Delphi, in model.py.

Yes, that's a typo, thanks for noticing! I'm fixing it right now.

> And I also believe the prediction should be using gender+age+logit to predict the next event.

Could you please clarify what you mean by this?
The next token is generated by predicting the rate (aka exp(logit)) using Delphi, based on the existing trajectory (which may or may not include a biological sex token) and the age of the last recorded token. Then the next token in sampled from the competing exponentials with the prediction rates (see header or https://github.com/gerstung-lab/Delphi/blob/main/sampling_trajectories.ipynb for more details)
